mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -804,6 +804,7 @@ mlir::scf::tileUsingSCFForallOp(RewriterBase &rewriter, TilingInterface op,
                                 const scf::SCFTilingOptions &options) {
   Location loc = op->getLoc();
   OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPointAfter(op);
```

Comment:
Thanks! I think this is needed cause we need to use some `tensor.dim` operations with the source being the result of `op`. Once those `tensor.dim` operations get resolved to operands this is not needed... If you have an example of where you hit this would be good to just add that. I do have a dynamic size test here, wondering why it didnt trigger there.